### ISSUE TITLE
Randomized fingerprints, but consistent?

### ISSUE THREAD

**Issue by AxbB36** [refraction-networking/utls#15]
`HelloRandomized` and friends provide a randomized fingerprint, but it is re-randomized for each new connection. Is there a way to get a "handle" to a specific randomized fingerprint so it can be re-used within a process?

Here's a sample program that makes 10 connections with `HelloRandomizedNoALPN`:
```
package main

import (
	"fmt"
	tls "github.com/refraction-networking/utls"
	"net"
)

const hostName = "golang.org"

func connect() error {
	conn, err := net.Dial("tcp", net.JoinHostPort(hostName, "443"))
	if err != nil {
		return err
	}
	uconn := tls.UClient(conn, nil, tls.HelloRandomizedNoALPN)
	uconn.SetSNI(hostName)
	err = uconn.Handshake()
	if err != nil {
		return err
	}
	return uconn.Close()
}

func main() {
	for i := 0; i < 10; i++ {
		err := connect()
		if err == nil {
			fmt.Println(i, "ok")
		} else {
			fmt.Println(i, err)
		}
	}
}
```
Sample output is:
```
0 ok
1 remote error: tls: handshake failure
2 ok
3 remote error: tls: handshake failure
4 ok
5 ok
6 ok
7 ok
8 ok
9 ok
```
I am using `tshark` to verify that the fingerprint changes with each connection:
```
tshark -V -f 'host golang.org' -Y ssl.handshake.ciphersuites
```

Incidentally, the reason this came up is because I'm trying to use `UClient` inside the `DialTLS` of [`http.Transport`](https://golang.org/pkg/net/http/#Transport) in order to get TLS fingerprint obfuscation along with connection reuse, keepalive, and the other features that `http.Transport` provides. When used with `utls`, `http.Transport` seems to create 10 independent connections (each with its own randomized fingerprint). I'll file a separate issue about that.

**Comment by AxbB36** [comment]
> When used with `utls`, `http.Transport` seems to create 10 independent connections (each with its own randomized fingerprint). I'll file a separate issue about that.

Never mind on this minor point. If I read and close the `Body` of each response, then `http.Transport` uses only one connection, even when HTTP/1.1 is negotiated. (When HTTP/2 is negotiated, it uses only one connection, whether the bodies are read and closed or not.)

**Comment by sergeyfrolov** [comment]
Implemented in #20
